src/utils/scope.js

```diff
@@ -0,0 +1,40 @@
+import { createEmptyMap, isMap } from './map.js'
+
+function assign (scope, ...objects) {
```

Comment:
I wondered about this. 

`assign` could live in `utils/object` or `utils/map`, though my preference is map: if it's analogue is `Object.assign(target, ...objs)`. This `assign`, `target` is always a Map, even though `objs` is `(Map|Object)[]`.

`createSubScope` does feels scope specific, which is why I put it here. It _should_ forward to the scope if it can.

Reply:
makes sense to rename the file `scope.js` to `map.js`, you're right! Maybe `createSubScope` can be renamed to `createSubMap` :)

Reply:
I've moved `assign` to `map.js`, and changed the body of `createSubScope`, and changed the `examples/custom_scope_object.js` example. Hopefully this should be instructive.